src/google-maps/map-polyline/map-polyline.ts

```diff
@@ -0,0 +1,234 @@
+  encapsulation: ViewEncapsulation.None,
+})
+export class MapPolyline implements OnInit, OnDestroy {
+  @Input()
+  set options(options: google.maps.PolylineOptions) {
+    this._options.next(options || {});
+  }
+
+  @Input()
+  set path(path: google.maps.MVCArray<google.maps.LatLng>|google.maps.LatLng[]|
+           google.maps.LatLngLiteral[]) {
+    this._path.next(path);
+  }
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.click
+   */
+  @Output() mapClick = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.dblclick
+   */
+  @Output() mapDblclick = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.drag
+   */
+  @Output() mapDrag = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.dragend
+   */
+  @Output() mapDragend = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.dragstart
+   */
+  @Output() mapDragstart = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mousedown
+   */
+  @Output() mapMousedown = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mousemove
+   */
+  @Output() mapMousemove = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseout
+   */
+  @Output() mapMouseout = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseover
+   */
+  @Output() mapMouseover = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseup
+   */
+  @Output() mapMouseup = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.rightclick
+   */
+  @Output() mapRightclick = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  private readonly _options = new BehaviorSubject<google.maps.PolylineOptions>({});
+  private readonly _path =
+      new BehaviorSubject<google.maps.MVCArray<google.maps.LatLng>|google.maps.LatLng[]|
+                          google.maps.LatLngLiteral[]|undefined>(undefined);
+
+  private readonly _destroy = new Subject<void>();
+
+  private readonly _listeners: google.maps.MapsEventListener[] = [];
+
+  _polyline?: google.maps.Polyline;
+
+  constructor(private readonly googleMap: GoogleMap) {}
```

Comment:
We prefix private properties with `_`. There's actually a lint rule for it, but it didn't catch this case for some reason. I'll look into why it wasn't caught. Also maybe this should be called just `_map`?

Reply:
Done.